lib/common.sh

```
@@ -0,0 +1,316 @@
+    # Create target directory if needed
+    local target_dir=$(dirname "$target")
+    if [ ! -d "$target_dir" ]; then
+        mkdir -p "$target_dir"
+        log_info "Created directory: $target_dir"
+    fi
+    
+    # Handle existing target
+    if [ -e "$target" ] || [ -L "$target" ]; then
+        if [ -L "$target" ]; then
+            # It's already a symlink
+            local current_source=$(readlink "$target")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)